### PR TITLE
Fix CTRL C Scoping

### PR DESCRIPTION
**Pull request by jdcaporale**
Addresses Issue #51. 
 
Moves the global variable into a single translation unit (common_header.cpp) and then accesses it via methods. 
Initial testing of this method worked, but this code remains untest and can be reviewed but DO NOT MERGE until tested. 

Also, handles a TODO that was stuck behind this fix, which moves LCM Subscriber source code into a cpp.

### CODE REVIEW

File: include/kodlab_mjbots_sdk/common_header.h [hunk 1]
```diff
@@ -9,18 +19,25 @@
...
  */
-static std::atomic_bool CTRL_C_DETECTED(false);
+void ActivateCtrlC();
+
+/**
+ * @brief Returns the state of the CtrlC Flag
+ */
+bool CtrlCDetected();
+
+// Ctrl-C signal handling adapted from solo8 repo
```
Comment (ShaneRozenLevy) [inline include/kodlab_mjbots_sdk/common_header.h:32]: Does this comment still belong here?

**Review by ShaneRozenLevy — APPROVED** [review]
LGTM. One comment seemed out of place.

File: include/kodlab_mjbots_sdk/common_header.h (hunk 1 above)
Comment (jdcaporale) [inline include/kodlab_mjbots_sdk/common_header.h:32]:
```suggestion
```

**Comment by jdcaporale** [comment]
Test and ready.

**Pull request merged**